src/ui/public/vis_maps/visualizations/marker_types/base_marker.js

```diff
@@ -229,8 +230,51 @@ export default function MarkerFactory() {
       this._createTooltip(content, latLng);
     }
 
+    _getOffset(content, latLng) {
+      // Default maxWidth leaflet applies to popup is 300
+      const popupWidth = 300;
+
+      // We need to create the popup first to determine how tall it will be.  Give
+      // it an out of map bounds offset as we don't want to see this one.  It will
+      // get replaced on the next popup creation where we apply the correct offset
+      const popup = L.popup({ autoPan: false, offset: new L.Point(this.map.getSize().x * 2, this.map.getSize().y * 2) })
+      .setLatLng(latLng)
+      .setContent(content)
+      .openOn(this.map);
+      const popupHeight = popup._contentNode.clientHeight;
```

Comment:
We probably shouldn't introduce dependencies on internal implementation details on 3rd party libraries.